python/lib/metrics/dmod/metrics/communication.py

```diff
@@ -278,9 +279,18 @@ def write(self, reason: REASON_TO_WRITE, data: dict, verbosity: Verbosity = None
             data:
             verbosity:
         """
-        for communicator in self.__communicators.values():
-            if not verbosity or verbosity and communicator.verbosity >= verbosity:
-                communicator.write(reason=reason, data=data)
+        try:
+            for communicator in self.__communicators.values():
+                if not verbosity or verbosity and communicator.verbosity >= verbosity:
+                    communicator.write(reason=reason, data=data)
+        except:
+            message = traceback.format_exc()
+            logging.error(message)
+
+            # The message is also printed since logging sometimes forces all newlines into a single line with just
+            # the "\n" character, making the error hard to read
+            print(message)
```

Comment:
Controversial approach: I'm leaving the `print` just because that `\n` rendering was a real problem without it, but I'm floating up the message in a new exception as suggested rather than just re-raising.

Reply:
Seems hacky and like something you would probably have to do to make python logging happy...

Reply:
Sorry for not reading your inline comment. I dont think I would have commented if I had.